lib/win32/certstore/store_base.rb

```diff
@@ -84,18 +84,23 @@ def cert_list(store_handler)
       # store_handler => Open certificate store handler
       # certificate_thumbprint => thumbprint is a hash. which could be sha1 or md5.
       def cert_delete(store_handler, certificate_thumbprint)
+        validate_thumbprint(certificate_thumbprint)
+        cert_name = memory_ptr
+        thumbprint = update_thumbprint(certificate_thumbprint)
+        cert_pem = get_cert_pem(thumbprint)
+        cert_pem = format_pem(cert_pem)
+        cert_rdn = get_rdn(build_openssl_obj(cert_pem))
+        result = false
         begin
-          if !certificate_name.empty? && (pcert_context = CertFindCertificateInStore(store_handler, X509_ASN_ENCODING, 0, CERT_FIND_ISSUER_STR, certificate_name.to_wstring, nil)) && (not pcert_context.null?)
+          cert_args = cert_find_args(store_handler, cert_rdn)
+          if pcert_context = CertFindCertificateInStore(*cert_args)
```

Comment:
I am not clear if your are trying to assign the value of `CertFindCertificateInStore(*cert_args)` to `pcert_context` or check for equality -- if the latter, then you need to change this to `==`

Reply:
@rhass, I know this is little bit confusing, but here `pcert_context` is pointer of desired certificate. `CertFindCertificateInStore` is basically searching a certificate in open certificate store and if that is found then `pcert_context` pointer applied, after that we can use this `pcert_context` to delete the certificate. For ref: https://msdn.microsoft.com/en-us/library/windows/desktop/aa376064(v=vs.85).aspx